extensions/s3/src/main/java/io/deephaven/extensions/s3/S3AsyncClientFactory.java

```diff
@@ -58,9 +56,27 @@ static S3AsyncClient getAsyncClient(@NotNull final S3Instructions instructions)
                         .scheduledExecutorService(ensureScheduledExecutor())
                         .build())
                 .credentialsProvider(instructions.awsV2CredentialsProvider());
-        instructions.regionName().map(Region::of).ifPresent(builder::region);
+        if (instructions.regionName().isPresent()) {
+            builder.region(Region.of(instructions.regionName().get()));
+        } else {
+            // If region is not provided, we enable cross-region access to allow the SDK to determine the region
+            // based on the bucket location and cache it for future requests.
+            builder.crossRegionAccessEnabled(true);
+        }
         instructions.endpointOverride().ifPresent(builder::endpointOverride);
-        final S3AsyncClient s3AsyncClient = builder.build();
+        S3AsyncClient s3AsyncClient;
+        try {
+            s3AsyncClient = builder.build();
+        } catch (final Exception e) {
+            if (e.getMessage().contains("region") && instructions.regionName().isEmpty()) {
+                // TODO Check with Devin if this feels okay, this seems hacky to me but I am not able to figure out a
```

Comment:
Check with Devin

Reply:
Can you provide the full error message returned? This seems at odds with the feature.

Reply:
Its the same error message you mentioned in the issue https://github.com/deephaven/deephaven-core/issues/6016
```
Value: failed to read parquet data. : RuntimeError: software.amazon.awssdk.core.exception.SdkClientException: Unable to load region from any of the providers in the chain software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain@3b64ffd5: [software.amazon.awssdk.regions.providers.SystemSettingsRegionProvider@6aa49bbc: Unable to load region from system settings. Region must be specified either via environment variable (AWS_REGION) or  system property (aws.region)., software.amazon.awssdk.regions.providers.AwsProfileRegionProvider@6cefdb33: No region provided in profile: default, software.amazon.awssdk.regions.providers.InstanceProfileRegionProvider@5fccb0f1: Unable to contact EC2 metadata service.]
Traceback (most recent call last):
  File "/Users/shivammalhotra/deephaven/projects/deephaven-core/py/server/deephaven/parquet.py", line 205, in read
    return Table(_JParquetTools.readTable(path, read_instructions))
RuntimeError: software.amazon.awssdk.core.exception.SdkClientException: Unable to load region from any of the providers in the chain software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain@3b64ffd5: [software.amazon.awssdk.regions.providers.SystemSettingsRegionProvider@6aa49bbc: Unable to load region from system settings. Region must be specified either via environment variable (AWS_REGION) or  system property (aws.region)., software.amazon.awssdk.regions.providers.AwsProfileRegionProvider@6cefdb33: No region provided in profile: default, software.amazon.awssdk.regions.providers.InstanceProfileRegionProvider@5fccb0f1: Unable to contact EC2 metadata service.]
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:111)
	at software.amazon.awssdk.regions.providers.AwsRegionProviderChain.getRegion(AwsRegionProviderChain.java:70)
	at software.amazon.awssdk.awscore.client.builder.AwsDefaultClientBuilder.resolveRegion(AwsDefaultClientBuilder.java:293)
	at software.amazon.awssdk.utils.AttributeMap$DerivedValue.primeCache(AttributeMap.java:600)
	at software.amazon.awssdk.utils.AttributeMap$DerivedValue.get(AttributeMap.java:589)
``